src/main/java/com/faforever/client/map/MapServiceImpl.java

```diff
@@ -167,6 +205,20 @@ protected Void call() throws Exception {
     }, callback);
   }
 
+  @Override
+  public List<Comment> getComments(String mapName) {
+    int id = getMapInfoBeanFromVaultFromName(mapName).getId();
+    if(id == 0){
+      return new ArrayList<>();
+    }
+    try {
+      return mapVaultParser.parseComments(id);
+    } catch (IOException e) {
+      logger.error("Error in parsing comment for {}", mapName);
+    }
+    return null;
```

Comment:
Shouldn't this return an empty list here as well? Also, use `Collections.emptyList()` instead of `new ArrayList<>()` (see line 212)